x86/CheckTime.asm

```diff
@@ -1,86 +1,104 @@
-         calign   16
 CheckTime:
-    ; we must:
-    ;   set signals stop if the search is to be aborted (soon)
-    ;   and determine a good resetCnt to send to all threads
-    ;     lower values of resetCnt lead to to better resolution but increased polling
-    ;     conversely for higher values of resetCnt
-           push   rbx rsi rdi
+        ; we must:
+        ;   set signals stop if the search is to be aborted (soon)
+        ;   and determine a good resetCnt to send to all threads
+        ;     lower values of resetCnt lead to to better resolution but increased polling
+        ;     conversely for higher values of resetCnt
+
+;ProfileInc CheckTime
+
+               push   rbx rsi rdi
+; AssertStackAligned   'CheckTime'
+
 	; if MAX_RESETCNT is exactly calls to search per second
 	; then this value of resetCnt should put us back here in 1 second
 	; this is obviously too much when using time mgmt
 	; so .Reset4Time reduces this
-            mov   esi, MAX_RESETCNT		; fall through count
-    ; Of course, setting MAX_RESETCNT = number of calls to search per second
-    ; is impossible to measure and also dangerous. So really we have
-    ;   MAX_RESETCNT ~= (number of calls to search per second) * X
-    ; where X is a number between 0 and 1.
-    ; We then expect to be back here in X seconds.
-            xor   eax, eax
-            cmp   al, byte[limits.ponder]
-            jne   .Return
-            cmp   al, byte[limits.useTimeMgmt]
-             je   .DontUseTimeMgmt
-            mov   rdi, qword[time.maximumTime]
+
+                mov   esi, MAX_RESETCNT		; fall through count
+
+        ; Of course, setting MAX_RESETCNT = number of calls to search per second
+        ; is impossible to measure and also dangerous. So really we have
+        ;   MAX_RESETCNT ~= (number of calls to search per second) * X
+        ; where X is a number between 0 and 1.
+        ; We then expect to be back here in X seconds.
+
+                xor   eax, eax
+                cmp   al, byte[limits.ponder]
+                jne   .Return
+
+                cmp   al, byte[limits.useTimeMgmt]
+                 je   .DontUseTimeMgmt
+                mov   rdi, qword[time.maximumTime]
+
 .Reset4Time:
-    ; rdi is target time
-           call   Os_GetTime
-            sub   rax, qword[time.startTime]
-            add   rax, 1
-    ; rax is elapsed time
-            sub   rdi, rax
-             js   .Stop
-    ; If rdi ms are remaining, attemp to put us back here in X*rdi/2 ms.
-    ; The values of rdi at this point are in geometric progression.
-    ; On tested machine, this ends a 'go movetime 10000' in 10000 ms
-    ; with approx 130 calls to CheckTime.
+        ; rdi is target time
+               call   Os_GetTime
+                sub   rax, qword[time.startTime]
+                add   rax, 1
+        ; rax is elapsed time
+                sub   rdi, rax
+                 jb   .Stop
```

Comment:
This was upgraded to a "js" to handle possible timer wrap around. "js" is better her.